include/picongpu/algorithms/AssignedTrilinearInterpolation.hpp

```diff
@@ -52,64 +52,69 @@ namespace picongpu
         /** Does a 3D trilinear field-to-point interpolation for
          * arbitrary assignment function and arbitrary field_value types.
          *
-         * @tparam T_AssignmentFunction function for assignment
          * @tparam T_begin lower margin for interpolation
          * @tparam T_end upper margin for interpolation
+         * @tparam T_AssignmentFunctionArray type of the array with shape functors
          *
          * @param cursor cursor pointing to the field
-         * @param pos position of the interpolation point
+         * @param shapeFunctorArray Array with d shape functors, where d is the number of simulation.
+         *                          The shape functor must have the interface to call `operator()(relative_grid_point)`
+         *                          and return the assignment value for the given grid point.
          * @return sum over: field_value * assignment
          *
          * interpolate on grid points in range [T_begin;T_end]
          */
-        template<typename T_AssignmentFunction, int T_begin, int T_end, typename T_Cursor>
-        HDINLINE static auto interpolate(const T_Cursor& cursor, const float3_X& pos) ->
-            typename ::pmacc::result_of::Functor<AssignedTrilinearInterpolation, T_Cursor>::type
+        template<int T_begin, int T_end, typename T_Cursor, typename T_AssignmentFunctionArray>
```

Comment:
It seems like `T_AssignmentFunctionArray` is actually a single functor type, so should be named `T_AssignmentFunction` ?